src/Helper/HttpClientFactory.php

```diff
@@ -64,6 +75,14 @@ public function createClient(array $config = [])
             $config['defaults']['cookies'] = $this->cookieJar;
         }
 
+        if (!isset($config['defaults']['headers'])) {
+            $config['defaults']['headers'] = [];
+        }
+        if (!isset($config['defaults']['headers']['User-Agent'])) {
+            // need to set the user-agent
+            $config['defaults']['headers']['User-Agent'] = $this->userAgent;
+        }
+
```

Comment:
Maybe this should be moved before line 67 so if admin didn't enabled paywalled website, they'll still got the default user agent.

Reply:
It's done